pika/adapters/amqp_connector.py

```diff
@@ -0,0 +1,50 @@
+"""Implements the workflow of performing multiple TCP/[SSL]/AMQP connection
+attempts with timeouts and retries until one succeeds or all attempts fail.
+
+"""
+
+class AMQPConnector(object):
+    """Implements the workflow of performing multiple TCP/[SSL]/AMQP connection
+    attempts with timeouts and retries until one succeeds or all attempts fail.
+
+    The workflow:
```

Comment:
Users have been asking for this for a while. This PR makes it possible to deliver such functionality.